controllers/modals/submissionPayments/SubmissionPaymentsHandler.inc.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/**
+ * @file controllers/modals/submissionPayments/SubmissionPaymentsHandler.inc.php
+ *
+ * Copyright (c) 2014-2020 Simon Fraser University
+ * Copyright (c) 2003-2020 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class SubmissionPaymentsHandler
+ * @ingroup controllers_modals_submissionPayments
+ *
+ * @brief A handler to manage publication fees.
+ */
+
+// Import the base Handler.
+import('classes.handler.Handler');
+
+class SubmissionPaymentsHandler extends Handler {
+
+	/** @var Submission **/
+	public $submission;
+
+	/**
+	 * Constructor.
+	 */
+	function __construct() {
+		parent::__construct();
+		$this->addRoleAssignment(
+			[ROLE_ID_SUB_EDITOR, ROLE_ID_MANAGER, ROLE_ID_ASSISTANT],
+			['status', 'save']
+		);
+	}
+
+
+	//
+	// Overridden methods from Handler
+	//
+	/**
+	 * @copydoc PKPHandler::initialize()
+	 */
+	function initialize($request) {
+		parent::initialize($request);
+		$this->submission = $this->getAuthorizedContextObject(ASSOC_TYPE_SUBMISSION);
+		$this->setupTemplate($request);
+	}
+
+	/**
+	 * @copydoc PKPHandler::authorize()
+	 */
+	function authorize($request, &$args, $roleAssignments) {
+		import('lib.pkp.classes.security.authorization.SubmissionAccessPolicy');
+		$this->addPolicy(new SubmissionAccessPolicy($request, $args, $roleAssignments));
+		return parent::authorize($request, $args, $roleAssignments);
+	}
+
+
+	//
+	// Public handler methods
+	//
+	/**
+	 * Display the status of publication payments.
+	 *
+	 * @param $args array
+	 * @param $request PKPRequest
+	 * @return JSONMessage JSON object
+	 */
+	public function status($args, $request) {
+		 AppLocale::requireComponents(LOCALE_COMPONENT_APP_EDITOR);
+
+		 $submissionPaymentsForm = new APP\components\forms\publication\SubmissionPaymentsForm(
+			$request->getDispatcher()->url($request, ROUTE_COMPONENT, null, 'modals.submissionPayments.SubmissionPaymentsHandler', 'save', null, ['submissionId' => $this->submission->getId()]),
+			$this->submission,
+			$request->getContext()
+		);
+
+		$templateMgr = TemplateManager::getManager($request);
+		$templateMgr->assign([
+			'assignData' => [
+				'components' => [
+					FORM_SUBMISSION_PAYMENTS => $submissionPaymentsForm->getConfig(),
+				],
+			],
+		]);
+
+		return $templateMgr->fetchJson('controllers/modals/submissionPayments/status.tpl');
+
+	}
```

Comment:
Rather than requiring a separate request to load this form, it should be moved into the root Workflow component handler. That way, the form can be toggled without any extra request to the server. You just need to move this code to `WorkflowHandler`. Here's an example of the issue entry form being loaded into the Workflow componpent.

- Instantiate the form: https://github.com/pkp/ojs/blob/master/pages/workflow/WorkflowHandler.inc.php#L73
- Pass the constant to the browser; https://github.com/pkp/ojs/blob/master/pages/workflow/WorkflowHandler.inc.php#L86
- Add the form as a managed component: https://github.com/pkp/ojs/blob/master/pages/workflow/WorkflowHandler.inc.php#L90-L91

Then in the template you can instantiate the form component directly (see common on workflow.tpl below).

Reply:
Done!